### ISSUE TITLE
Previous Window (netrw-P) often opens "empty" buffers

### ISSUE THREAD

**Issue by venthur** [neovim/neovim#23597]
### Problem

Often (but not always!) shows an empty buffer when I open a file from netrw using "P" (netrw-P) to open it in a previous window. With empty buffer I mean that the file is shown with no content in the previous buffer although the file does contain text.

### Steps to reproduce

nvim --clean
:Explore
* open some file with "v"> to create the "previous buffer"
* go back to netrw browse the directory tree and try to open some files using the "P" shortcut

It seems to be more reproducible if you try to open files that reside in different directories.

### Expected behavior

the files should always open in the window created by the first "v" command.

### Neovim version (nvim -v)

0.7.2

### Vim (not Nvim) behaves the same?

yes, 9.0

### Operating system/version

Linux

### Terminal name/version

gnome-terminal

### $TERM environment variable

xterm-256color

### Installation

Debian

**Comment by clason** [comment]
Please try the latest nightly (can be downloaded from our releases page).

**Comment by venthur** [comment]
Still the same with NVIM v0.10.0-dev-338+g189e21ae5 -- mind that the netrw version is still the same v171.

**Comment by clason** [comment]
> Still the same with NVIM v0.10.0-dev-338+g189e21ae5 -- mind that the netrw version is still the same v171.

Yes, that is the commit just _before_ the netRW version update. Please try again with latest master.

**Comment by venthur** [comment]
Done, same issue with `v0.10.0-dev-340+g33687f5e8` and `v173`. 

My vimscript is really bad, but I suspect is has something to do with `b:netrw_curdir` not being set correctly.

**Comment by dusanbrankov** [comment]
I'm having the same problem, just tried it with the [latest nightly version](https://github.com/neovim/neovim/releases/tag/nightly).

However, with `netrw_browse_split = 4`, it works normally with the `return` key, even when opening a file outside the working directory.

```lua
-- when browsing, <cr> will open the file by:
-- =0: re-using the same window  (default)
-- =1: horizontally splitting the window first
-- =2: vertically   splitting the window first
-- =3: open file in new tab
-- =4: act like "P" (ie. open previous window)
vim.g.netrw_browse_split = 4
```

**Comment by lucitez** [comment]
I'm having the same issues.

To reproduce `netrw-P` opening empty buffers:
- open nvim on directory A
- open a file using `<Cr>`
- open `:Vex:`
- open a file in directory A/B using `netrw-P` <-- this works
- switch back to directory browser
- open a file in directory A/C using `netrw-P` <-- this opens an empty buffer

To reproduce the :Lex error
- open nvim on a directory
- open a file using `<Cr>`
- open `:Lex`
- close the window containing the opened file (at this point, the directory browser is the only window open)
- try to open a file using `<Cr>` <-- this produces the following error:
```
Error detected while processing function <SNR>31_NetrwBrowseChgDir:line  142:
E471: Argument required: keepj keepalt 2wincmd 1
```

Since both :Vex and :Lex have issues, I feel like there isn't a great way to explore a file system with a split pane.

**Comment by lucitez** [comment]
As for the :Lex error, I believe I've found the source of the bug. Using :Lex sets g:netrw_chgwin to 2, but then if you close the second window the netrw_chgwin stays at 2. So then if you try to `<Cr>` on a file in explorer, netrw tries to open it in a second window which doesn't exist anymore. I think the solution would be to set `g:netrw_chgwin = -1` when closing the second window.

[:help netrw-C](https://macvim.org/docs/pi_netrw.txt.html#netrw-C)
